Logger.Tests/BaseLoggerMixinsTests.cs

```diff
@@ -1,32 +1,131 @@
 ﻿using Microsoft.VisualStudio.TestTools.UnitTesting;
 using System;
+using System.IO;
 using System.Collections.Generic;
+using Logger;
 
-namespace Logger.Tests;
-
-[TestClass]
-public class BaseLoggerMixinsTests
+namespace Logger.Tests
 {
+    [TestClass]
+    public class BaseLoggerMixinsTests
+    {
+
+        [TestMethod]
+        [ExpectedException(typeof(ArgumentNullException))]
+        public void Error_WithNullLogger_ThrowsException()
+        {
+            // Arrange
+            var logger = new TestLogger();
+
+            // Act
+            logger.Error(null, 0);
+
+            // Assert
+        }
+
+        [TestMethod]
+        public void Error_WithData_LogsMessage()
+        {
+            // Arrange
+            var logger = new TestLogger();
+
+            // Act
+            logger.Error("Message {0}", 42);
+
+            // Assert
+            Assert.AreEqual(1, logger.LoggedMessages.Count);
+            Assert.AreEqual(LogLevel.Error, logger.LoggedMessages[0].LogLevel);
+            Assert.AreEqual("Message 42", logger.LoggedMessages[0].Message);
+        }
+
+    
+
+
+
+    [TestMethod]
+    [ExpectedException(typeof(ArgumentNullException))]
+    public void Warning_WithNullLogger_ThrowsException()
+    {
+        // Arrange
+        var logger = new TestLogger();
+
+        // Act
+        logger.Warning(null, 0);
+
+        // Assert
+    }
+
+    [TestMethod]
+    public void Warning_WithData_LogsMessage()
+    {
+        // Arrange
+        var logger = new TestLogger();
+
+        // Act
+        logger.Warning("Message {0}", 42);
+
+        // Assert
+        Assert.AreEqual(1, logger.LoggedMessages.Count);
+        Assert.AreEqual(LogLevel.Error, logger.LoggedMessages[0].LogLevel);
+        Assert.AreEqual("Message 42", logger.LoggedMessages[0].Message);
+    }
+
+
+
+
+    [TestMethod]
+    [ExpectedException(typeof(ArgumentNullException))]
+    public void Information_WithNullLogger_ThrowsException()
+    {
+        // Arrange
+        var logger = new TestLogger();
+
+        // Act
+        logger.Information(null, 0);
+
+        // Assert
+    }
+
+    [TestMethod]
+    public void Information_WithData_LogsMessage()
+    {
+        // Arrange
+        var logger = new TestLogger();
+
+        // Act
+        logger.Information("Message {0}", 42);
+
+        // Assert
+        Assert.AreEqual(1, logger.LoggedMessages.Count);
+        Assert.AreEqual(LogLevel.Error, logger.LoggedMessages[0].LogLevel);
```

Comment:
For each test, you asserted that the log level type was `LogLevel.Error`, which made your tests pass but your functionality be incorrect.
```suggestion
        Assert.AreEqual(LogLevel.Information, logger.LoggedMessages[0].LogLevel);
```